internal/monitors/conviva/accountservice.go

```diff
@@ -0,0 +1,225 @@
+package conviva
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+//const (
+//	metricLensURLFormat = "https://api.conviva.com/insights/2.4/metrics.json?metrics=%s&account=%s&filter_ids=%s&metriclens_dimension_id=%d"
+//)
+
+var anyErrors bool
+var isInitialized bool
+
+// Account for Conviva account data
+type account struct {
+	id                   string
+	Name                 string
+	filters              map[string]string
+	metricLensFilters    map[string]string
+	metricLensDimensions map[string]float64
+}
+
+// AccountService interface for Account related methods
+type accountService interface {
+	getDefault()                                                             *account
+	getMetricLensDimensionMap(accountName string)                            map[string]float64
+	getID(accountName string)                                                string
+	getFilters(accountName string)                                           map[string]string
+	getMetricLensFilters(accountName string)                                 map[string]string
+	getFilterID(accountName string, filterName string)                       string
+	getMetricLensDimensionID(accountName string, metricLensDimension string) float64
+}
+
+type accountServiceImpl struct {
+	defaultAccount *account
+	accounts       []*account
```

Comment:
Why don't you just make `accounts` a `map[string]*account` and make the key the account name so that you don't have to do all of those loops in the get* methods?